packages/edit-site/src/components/layout/index.js

```diff
@@ -174,6 +174,16 @@ export default function Layout( { route } ) {
 
 					{ isMobileViewport && areas.mobile && (
 						<div className="edit-site-layout__mobile">
+							{ canvasMode !== 'edit' && (
+								<SidebarContent routeKey={ routeKey }>
```

Comment:
`SidebarContent` is adding a padding that makes the logo jump to the left when we navigate from the general site editor into the templates view. The class applying the padding is `.edit-site-sidebar__screen-wrapper`

Reply:
👍

Fixed as of https://github.com/WordPress/gutenberg/pull/63118/commits/29123181d26357ab13b97be2d256cef124b70437